protocol/ticket.re

```diff
@@ -0,0 +1,28 @@
+open Tezos_interop;
+
+// TODO: I feel uneasy about this module
```

Comment:
Why do you feel uneasy about this module?

Reply:
I'm also reminded of functors like `Set_with_yojson_make` and `Map_with_yojson_make` - can they be unified?

Reply:
(feel free to resolve)

Reply:
A lot of duplicated code and re exporting a type from Tezos_interop, it's probably a bad abstraction. Also nope, `Set_with_yojson_make` is completely different from `Map_with_yojson_make`.

Reply:
It's already slightly improved on a future PR tho